src/System.Globalization/tests/CharUnicodeInfo/CharUnicodeInfoTestData.cs

```diff
@@ -0,0 +1,224 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+
+namespace System.Globalization.Tests
+{
+    public static class CharUnicodeInfoTestData
+    {
+        private static List<CharUnicodeInfoTestCase> s_testCases;
+        public static List<CharUnicodeInfoTestCase> TestCases
+        {
+            get
+            {
+                if (s_testCases == null)
+                {
+                    s_testCases = new List<CharUnicodeInfoTestCase>(); 
+                    string fileName = "UnicodeData.txt";
+                    Stream stream = typeof(CharUnicodeInfoGetUnicodeCategoryTests).GetTypeInfo().Assembly.GetManifestResourceStream(fileName);
+                    using (StreamReader reader = new StreamReader(stream))
+                    {
+                        while (!reader.EndOfStream)
+                        {
+                            s_testCases.Add(Parse(reader.ReadLine()));
+                        }
+                    }
+                }
+
+                return s_testCases;
+            }
+        }
+
+        private static CharUnicodeInfoTestCase Parse(string line)
+        {
+            // Data is in the format: 
+            // code-value;
+            // character-name; (ignored)
+            // general-category;
+            // canonical-combining-classes; (ignored)
+            // bidirecional-category; (ignored)
+            // character-decomposition-mapping; (ignored)
+            // decimal-digit-value; (ignored)
+            // digit-value; (ignoed)
+            // number-value;
+            string[] data = line.Split(';');
+            string charValueString = data[0];
+            string charCategoryString = data[2];
+            string numericValueString = data[8];
+
+            int codeValue = int.Parse(charValueString, NumberStyles.HexNumber);
+            string codeValueRepresentation = codeValue > char.MaxValue ? char.ConvertFromUtf32(codeValue) : ((char)codeValue).ToString();
```

Comment:
I couldn't use [MemberData], only [Fact] because the list of data contains invalid data